src/items.rs

```diff
@@ -127,8 +128,15 @@ impl Rewrite for ast::Local {
 
             if let Some(block) = else_block {
                 let else_kw_span = init.span.between(block.span);
+                // Strip attributes and comments to check if newline is needed before the else
+                // keyword from the initializer part. (#5901)
+                let init_str = if context.config.version() == Version::One {
+                    result.as_str()
+                } else {
+                    &result[let_kw_offset..]
+                };
                 let force_newline_else = pat_str.contains('\n')
-                    || !same_line_else_kw_and_brace(&result, context, else_kw_span, nested_shape);
+                    || !same_line_else_kw_and_brace(init_str, context, else_kw_span, nested_shape);
```

Comment:
There are two calls bellow that I think we need to change as well.

The `available_space` calculation should also be derived using the len without any leading attributes or comments. The call to `allow_single_line_let_else_block` should also take a string starting at the `let` kw.